gdxsv/lbs_handler.go

```diff
@@ -629,8 +630,11 @@ var _ = register(lbsPostGameParameter, func(p *LbsPeer, m *LbsMessage) {
 var _ = register(lbsAskKDDICharges, func(p *LbsPeer, m *LbsMessage) {
 	// 課金予測情報 (円)
 	p.SendMessage(NewServerAnswer(m).Writer().Write32(0).Msg())
-	var mcsUsersCount uint32 = uint32(len(sharedData.mcsUsers))
-	p.SendMessage(NewServerNotice(lbsBattleUserCount).Writer().Write32(mcsUsersCount).Msg())
+	go func() {
+		time.Sleep(1 * time.Second)
+		var mcsUsersCount uint32 = uint32(len(sharedData.mcsUsers))
+		p.SendMessage(NewServerNotice(lbsBattleUserCount).Writer().Write32(mcsUsersCount).Msg())
```

Comment:
The problem is accessing mcsUsers wihout locking mutex.

Reply:
I was using direct int value when testing this race condition, but yes I can improve to use the function call with locked mutex instead

Reply:
Updated

Reply:
Why need `go func() { time.Sleep(1 * time.Second) ` ? I hate that.

Reply:
The race condition exists within Flycast / Gundam, so I’m delaying 1 second to send the user count after KDDI charges, to make sure the `lbsBattleUserCount` value can override the KDDI value

Reply:
Is using `time.AfterFunc` preferred over go routine + sleep in that case?

Reply:
Simply send the command another timing. (eg, lbsServerMoney, lbsStartLobby)

Reply:
But how to ensure when the game request KDDI charges, it would be overridden by other api calls?

I didn't verify if the KDDI charges call only happen once after entering plaza

Reply:
Just ignore KDDICharge message if patch enabled.

Reply:
That’s a bit tricky since if the player opt out the langmod (Set it to disabled), the KDDI modem fee texture won’t be patched and the battle count is ignored.

And currently we are sending the same “Japanese” string for Japanese langmod and Disabled langmod users

But I would look into it, let’s hold this PR, thank you